code/src/App.js

```diff
@@ -1,9 +1,71 @@
-import React from 'react'
+import React, { useState, useEffect } from "react";
+import { ThoughtsCard } from "components/ThoughtsCard";
+import { Form } from "components/Form";
 
 export const App = () => {
+  const [thoughts, setThoughts] = useState([]);
+  const [clickCounts, setClickCounts] = useState(0);
+  const FETCH_URL = "https://happy-thoughts-technigo.herokuapp.com/thoughts";
+
+  useEffect(() => {
+    //Fetches data from the API: an array including all Thoughts and
+    //uses the setThoughts setter function to assign that data to the
+    //thoughts variable
+    fetch(FETCH_URL)
+      .then((response) => response.json())
+      //json is now the variable containing thoughts array we got from the API
+      .then((json) => {
+        //Filter only Thoughts that have some text in it
+        const filteredThoughts = json.filter((thought) => {
+          return thought.message !== "";
```

Comment:
It's not 100% necessary, our API is not allowing to POST empty messages either way.
On the other hand, it's good to have couple of safe checks, so at the end this code can stay :)